src/App.jsx

```diff
@@ -1,28 +1,93 @@
 import React from 'react';
 import './App.scss';
 import './styles/general.scss';
+import getResponse from './api/api';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
+const url = 'https://mate-api.herokuapp.com/todos';
+
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    titleFilter: '',
+    todoStatus: 'all',
   };
 
+  componentDidMount() {
+    getResponse(url)
+      .then((todos) => {
+        this.setState({ todos });
+      });
+  }
+
+  selectUser = (userId, event) => {
+    if (userId !== this.state.selectedUserId) {
+      this.setState({ selectedUserId: userId });
+    }
+  }
+
+  clearUser = () => {
+    this.setState({ selectedUserId: 0 });
+  }
+
+  inputHandler = (event) => {
+    this.setState({ titleFilter: event.target.value });
+  }
+
+  selectHandler = (event) => {
+    this.setState({ todoStatus: event.target.value });
+  }
+
   render() {
-    const { todos, selectedUserId } = this.state;
+    const { todos, selectedUserId, titleFilter, todoStatus } = this.state;
+
+    const formatedList = todos
+      .filter(todo => !Object.values(todo).includes(null))
+      .filter(todo => todo.title.includes(titleFilter))
+      .filter((todo) => {
+        switch (todoStatus) {
+          case ('all'):
+            return true;
+          case (todo.completed.toString()):
+            return true;
+          default:
+            return false;
+        }
+      });
 
     return (
       <div className="App">
+        <div className="searchWrap">
+          <label htmlFor="">
+            TODO name:
+            <input
+              type="text"
+              value={titleFilter}
+              onChange={this.inputHandler}
+            />
+          </label>
+          <select onChange={this.selectHandler}>
+            <option value="all">All</option>
+            <option value>Finished</option>
+            <option value={false}>Unfinished</option>
+          </select>
```

Comment:
I think this code should be placed in the TodoList component
![image](https://user-images.githubusercontent.com/77124061/120470122-dc6eb980-c3ab-11eb-9e4a-49ae647da47e.png)